src/server/api/resolvers/ballotResolver.ts

```diff
@@ -0,0 +1,42 @@
+import { Action } from '../../util/Permissions';
+import { BallotService } from '../../controllers/BallotService';
+import { ProjectScores } from '../types/ballot';
+
+
+const resolvers = {
+    Query: {
+        nextBallotSet: async (obj: any,
+                              args: { user_id?: number },
+                              context: any) => {
+            if (!context.user ||
+                !context.user.can(Action.ViewBallot, args.user_id)) {
+                throw new Error('You do not have permission to view ballots');
+            }
+
+            return BallotService.getNextProject(args.user_id!);
+        },
+    },
+
+    Mutation: {
+        scoreProject: (obj: any,
+                       args: { user_id?: number, scores?: ProjectScores[] },
+                       context: any) => {
+            if (!context.user ||
+                !context.user.can(Action.ScoreBallot, args.user_id)) {
+                throw new Error('You do not have permission to score ballots');
+            }
+
+            const ret = BallotService.scoreProject(args.user_id!,
```

Comment:
We'll have to look into Apollo typings at some point because we can't apply types to any of the resolver arguments. All attributes in the arguments must be optional so each argument fits under 'any'